jgrapht-core/src/main/java/org/jgrapht/alg/intervalgraph/IntervalGraphRecognizer.java

```diff
@@ -0,0 +1,273 @@
+
+        // Additionally, calculate the index and the corresponding B set for each vertex
+
+        // Step 6 - LBFS* with the resulting sweeps
+        // Input - the result of sweep gamma and sweep epsilon
+        // Output - the result of current sweep zeta
+        HashMap<V, Integer> sweepZeta =
+            lexBreadthFirstSearchStar(graph, vertexD, sweepDelta, sweepEpsilon);
+
+        // if sweepZeta is umbrella-free, then the graph is interval.
+        // otherwise, the graph is not interval
+
+        if (isIOrdering(sweepZeta, graph)) {
+            this.isIntervalGraph = true;
+
+            // Compute interval representation -- TODO: complete after merge
+            HashMap<V, Integer> neighborIndex = new HashMap<>();
+            for (V vertex : graph.vertexSet()) {
+                int maxNeighbor = 0;
+
+                List<V> neighbors = Graphs.neighborListOf(graph, vertex);
+                neighbors.add(vertex);
+
+                for (V neighbor : neighbors) {
+                    maxNeighbor = Math.max(maxNeighbor, sweepZeta.get(neighbor));
+                }
+
+                neighborIndex.put(vertex, maxNeighbor);
+            }
+
+            HashMap<Integer, Interval<Integer>> intervals = new HashMap<>(graph.vertexSet().size());
+            ArrayList<Interval<Integer>> sortedIntervals =
+                new ArrayList<>(graph.vertexSet().size());
+
+            // Compute intervals and store them associated by their starting point ...
+            for (V vertex : graph.vertexSet()) {
+                Interval<Integer> vertexInterval =
+                    new Interval<>(sweepZeta.get(vertex), neighborIndex.get(vertex));
+
+                intervals.put(sweepZeta.get(vertex), vertexInterval);
+            }
+
+            // ... and produce a list sorted by the starting points for an efficient construction of
+            // the graph
+            for (int i = 0; i < graph.vertexSet().size(); i++) {
+                sortedIntervals.add(intervals.get(i));
+            }
+
+            // TODO: build the actual interval graph
+            this.intervalRepresentation = null;
+        } else {
+            // set values negatively
+            this.isIntervalGraph = false;
+            this.intervalRepresentation = null;
+        }
+
+        return isIOrdering(sweepZeta, graph);
+    }
+    
+    /**
+     * Calculates if the given sweep is an I-Ordering (according to the Graph graph) in linear time.
+     * 
+     * @param <E>
+     *
+     * @param sweep the order we want to check if its an I-Order
+     * @param graph the graph we want to check if its an I-Order
+     * @return true, if sweep is an I-Order according to graph
+     */
+    private static <V, E> boolean isIOrdering(HashMap<V, Integer> sweep, Graph<V, E> graph)
+    {
+        // Compute inverse sweep map to quickly find vertices at given indices
+        HashMap<Integer, V> inverseSweep = new HashMap<>();
```

Comment:
This can be replaced by an array.

Reply:
Arrays of generics, unfortunately, are a bit of a pain in the ass in Java, which is why I opted to use HashMaps instead whenever I needed an array.

Reply:
Instead of arrays you can always use an appropriately initialized ArrayList